common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationSupport.java

```diff
@@ -500,9 +502,12 @@ public static UTF8String exec(final UTF8String srcString) {
     }
     public static UTF8String exec(
         final UTF8String srcString,
-        final UTF8String trimString,
+        UTF8String trimString,
         final int collationId) {
       CollationFactory.Collation collation = CollationFactory.fetchCollation(collationId);
+      if(collation.supportsSpaceTrimming) {
+        trimString = CollationFactory.applyTrimmingPolicy(trimString, collationId);
```

Comment:
I don't think this is the behaviour that we want - iiuc, you're trimming away spaces from the trimString (i.e. the users specifies that they want to trim spaces, and you revoke that given the collation spec - which is incorrect)

Reply:
note that all of these:

- `trim(..., "xy")`
- `trim(..., "xy ")`
- `trim(..., "x y")`
- `trim(..., " xy")`

should generally return exactly the same result (regardless of the collation, as long as they all use the same collation of course)

Reply:
Oh I see, you are right..

Reply:
so here are some examples to sum it all up:
- `trim("abcxyyxx", "xy")` -> should return: "abc" // suppose: UTF8_BINARY
- `trim("abcxyyxx    ", "xy")` -> should return: "abc    " // suppose: UTF8_BINARY_RTRIM
- `trim("abcxyyxx    ", "xy ")` -> should return: "abc" // suppose: UTF8_BINARY
- `trim("abcxyyxx    ", "xy ")` -> should return: "abc" // suppose: UTF8_BINARY_RTRIM

Reply:
Should be fixed, thanks.